### DISCUSSION TITLE
Lazer refuses to show a certain beatmap (Cold As Ice)

### DISCUSSION THREAD

**Discussion by toodols** [ppy/osu#22752] [Q&A] [answered]
**In short:** this realm file [client.realm.zip (old)](https://github.com/ppy/osu/files/10847474/client.realm.zip) won't show [Cold As Ice](https://osu.ppy.sh/beatmapsets/358119) in my beatmaps but if I delete (rename) client.realm and redownload it from the beatmap listings it works [client.backup.realm.zip (new)](https://github.com/ppy/osu/files/10847563/client.backup.realm.zip) but I can't bring myself to erase all my beatmaps and scores so I am looking for a resolution.

**Long version:** I noticed I didn't have Cold As Ice which I had way back from stable I tried loading it by dragging the .osz file and from osu.ppy.sh and it led to **Image B**. I tried searching it up in the beatmap listings (**Image C**), and after dealing with featured artists problem, it is the same problem.

I deleted (backed up) AppData\Roaming\osu, which worked (**Image D**) and slowly replaced files from the original backup until the beatmap refused to show again, and identified one singular file `client.realm` as the problem.

I did try to somewhat compare the original and the new one using [Realm Studio](https://docs.realm.io/sync/realm-studio/), but ran into issues
1. lazer's realm files are outdated and realm studio refuses to open them without "updating it to the latest version", which lazer deems as corrupt.
2. realm query language makes everything a pain to search through
3. i don't know what i'm looking for and i don't know what's wrong

Not an expert, not trying to be one, just want to play my favorite beatmap 

### (Image A) Current predicament - Searching "cold as ice" shows no results even though I have it downloaded
![image](https://user-images.githubusercontent.com/67543533/221783773-5a3aa0c8-7198-46be-866c-ab5e2ddcc0ae.png)

### (Image B) Dragging .osz file downloaded from osu.ppy.sh and clicking on the notification shows it on the left but still no beatmap somehow
![image](https://user-images.githubusercontent.com/67543533/221784079-ebca60bd-a36d-4e70-9133-4531231f9d41.png)

### (Image C) The beatmap is visible in beatmap listings and shows that I have it downloaded.
![image](https://user-images.githubusercontent.com/67543533/221785344-ab598293-b475-480b-b6d5-8b4dde8fdc73.png)

### (Image D) Deleting client.realm and reimporting the beatmap fixes everything but now I lose all my original beatmaps and scores
![image](https://user-images.githubusercontent.com/67543533/221782271-95b0c6d1-ee46-47ba-ba54-3f6267b64b65.png)

## Logs for broken client.realm one
[network.log](https://github.com/ppy/osu/files/10847593/network.log)
[performance.log](https://github.com/ppy/osu/files/10847594/performance.log)
[runtime.log](https://github.com/ppy/osu/files/10847595/runtime.log)
[updater.log](https://github.com/ppy/osu/files/10847596/updater.log)
[database.log](https://github.com/ppy/osu/files/10847597/database.log)
[input.log](https://github.com/ppy/osu/files/10847599/input.log)

**Comment by Theighlin** [comment, marked as answer]
Does changing the filter from date ranked and date submitted fix it? Those might not be populated

**Comment by Loreos7** [comment]
https://github.com/ppy/osu/issues/22416

**Comment by toodols** [comment]
Sorting by date added was the last thing I would ever think to try. That works. I think it's enough of a solution for now.
